File: backend/src/main/java/com/wecp/progressive/entity/Clinic.java
package com.wecp.progressive.entity;

public class Clinic {
    private int clinicId;
    private String clinicName;
    private String location;
    private int doctorId;
    private long contactNumber;
    private int establishedYear;

    public Clinic() {
    }

    public Clinic(int clinicId, String clinicName, String location, int doctorId, long contactNumber,
            int establishedYear) {
        this.clinicId = clinicId;
        this.clinicName = clinicName;
        this.location = location;
        this.doctorId = doctorId;
        this.contactNumber = contactNumber;
        this.establishedYear = establishedYear;
    }
    
    public int getClinicId() {
        return clinicId;
    }
    public void setClinicId(int clinicId) {
        this.clinicId = clinicId;
    }
    public String getClinicName() {
        return clinicName;
    }
    public void setClinicName(String clinicName) {
        this.clinicName = clinicName;
    }
    public String getLocation() {
        return location;
    }
    public void setLocation(String location) {
        this.location = location;
    }
    public int getDoctorId() {
        return doctorId;
    }
    public void setDoctorId(int doctorId) {
        this.doctorId = doctorId;
    }
    public long getContactNumber() {
        return contactNumber;
    }
    public void setContactNumber(long contactNumber) {
        this.contactNumber = contactNumber;
    }
    public int getEstablishedYear() {
        return establishedYear;
    }
    public void setEstablishedYear(int establishedYear) {
        this.establishedYear = establishedYear;
    }
    
}

File: frontend/src/app/mediconnect/models/Clinic.ts
import { Doctor } from './Doctor';

export class Clinic {
    clinicId: number;
    clinicName: string;
    location: string;
    contactNumber: string;
    establishedYear: number;
    doctor: Doctor;

    constructor(
        clinicId: number,
        clinicName: string,
        location: string,
        contactNumber: string,
        establishedYear: number,
        doctor: Doctor
    ) {
        this.clinicId = clinicId;
        this.clinicName = clinicName;
        this.location = location;
        this.contactNumber = contactNumber;
        this.establishedYear = establishedYear;
        this.doctor = doctor;
    }
}
